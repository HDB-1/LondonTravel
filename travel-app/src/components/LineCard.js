import React, { Component } from 'react'
import '../styling/LineCard.css'


export default class Items extends Component {

    render() {
        let { name, colour, status, reason } = this.props;
        return (
            <div className="card">
                <div className="card-header" style={colour}>
                    <h3>{name}</h3>
                </div>
                <div className="card-body">
                    <h5>{status}</h5>
                    {reason && <p>{reason}</p>}
                </div> 
                <div className="card-footer"><br/></div>
            </div>
        )
    }
}
