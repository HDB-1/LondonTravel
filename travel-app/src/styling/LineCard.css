div[class="card"] {
    display: flex;
    flex-direction: column;
    justify-content: center;
    margin-right: 10px;
    margin-bottom: 10px;
    width: 400px;
    border: 2px solid black;
}

div[class="card-header"] {
    text-align: center;
}