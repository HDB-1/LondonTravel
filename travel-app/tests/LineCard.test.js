import React from 'react';
import { shallow } from 'enzyme';
import LineCard from '../src/components/LineCard';

describe("shallow LineCard", () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<LineCard />)
    })
    it("should render correctly", () => {
        expect(wrapper).toMatchSnapshot();
    })
    
})