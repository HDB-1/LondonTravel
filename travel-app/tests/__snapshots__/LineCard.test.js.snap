// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow LineCard should render correctly 1`] = `
<div
  className="card"
>
  <div
    className="card-header"
  >
    <h3 />
  </div>
  <div
    className="card-body"
  >
    <h5 />
  </div>
  <div
    className="card-footer"
  >
    <br />
  </div>
</div>
`;
